### PR TITLE
Add dataset for Financial PhraseBank

### PR DESCRIPTION
**Pull request by frankier**

### CODE REVIEW

File: datasets/financial_phrasebank/README.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+
```
Comment (lhoestq) [inline datasets/financial_phrasebank/README.md:1]:
```suggestion
```

File: datasets/financial_phrasebank/README.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+- sentence: a tokenized line from the dataset
+- label: a label corresponding to the class as a string: 'positive', 'negative' or 'neutral'
+
+### Data Splits
+
+`sentences_50agree`; Number of instances with >=50% annotator agreement: 4846 
+`sentences_66agree`: Number of instances with >=66% annotator agreement: 4217
+`sentences_75agree`: Number of instances with >=75% annotator agreement: 3453
+`sentences_allagree`: Number of instances with 100% annotator agreement: 2264
```
Comment (lhoestq) [inline datasets/financial_phrasebank/README.md:90]:
```suggestion
There's no train/validation/test split.

However the dataset is available in four possible configurations depending on the percentage of agreement of annotators:

`sentences_50agree`; Number of instances with >=50% annotator agreement: 4846 
`sentences_66agree`: Number of instances with >=66% annotator agreement: 4217
`sentences_75agree`: Number of instances with >=75% annotator agreement: 3453
`sentences_allagree`: Number of instances with 100% annotator agreement: 2264
```

File: datasets/financial_phrasebank/financial_phrasebank.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "sentence": datasets.Value("string"),
+                    "label": datasets.features.ClassLabel(
+                        names=[
+                            "positive",
+                            "neutral",
+                            "negative",
+                        ]
```
Comment (lhoestq) [inline datasets/financial_phrasebank/financial_phrasebank.py:124]:
```suggestion
                        names=[
                            "negative",
                            "neutral",
                            "positive",
                        ]
```
usually we set "negative" at index 0 (by convention)

You'll need to regenerate the dataset_infos.json after this change

File: datasets/financial_phrasebank/README.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+### Citation Information
+
+@article{Malo2014GoodDO,
+  title={Good debt or bad debt: Detecting semantic orientations in economic texts},
+  author={P. Malo and A. Sinha and P. Korhonen and J. Wallenius and P. Takala},
+  journal={Journal of the Association for Information Science and Technology},
+  year={2014},
+  volume={65}
+}
```
Comment (lhoestq) [inline datasets/financial_phrasebank/README.md:190]:
````suggestion
```
@article{Malo2014GoodDO,
  title={Good debt or bad debt: Detecting semantic orientations in economic texts},
  author={P. Malo and A. Sinha and P. Korhonen and J. Wallenius and P. Takala},
  journal={Journal of the Association for Information Science and Technology},
  year={2014},
  volume={65}
}
```
````
use a code block for the citation

**Review by lhoestq — COMMENTED** [review]
This is really cool thank you !
I left a few comments

**Comment by frankier** [comment]
Thanks for the feedback. All accepted and metadata regenerated.

**Review by lhoestq — APPROVED** [review]
Thanks for the changes !
This looks all good now :)

**Pull request merged**